pysite/mixins.py

```diff
@@ -59,3 +59,25 @@ def table(self) -> Table:
     @property
     def db(self) -> RethinkDB:
         return self._db()
+
+
+class OauthMixin:
+
+    @classmethod
+    def setup(cls: "DBMixin", manager: "pysite.route_manager.RouteManager", blueprint: Blueprint):
+
+        if hasattr(super(), "setup"):
+            super().setup(manager, blueprint)  # pragma: no cover
+
+        cls._oauth = ref(manager.oauth_backend)
```

Comment:
I still think we might want to be able to access this class directly, so a property like the following might be nice:

```py
@property
def oauth(self):
    return self._oauth()
```